<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>833</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FSM Tool</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3" stretch="4,0">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,0">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QStackedWidget" name="leftMenu">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="currentIndex">
           <number>3</number>
          </property>
          <widget class="QWidget" name="fsmSettings">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>40</y>
              <width>225</width>
              <height>351</height>
             </rect>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <property name="verticalSpacing">
              <number>3</number>
             </property>
             <item row="4" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_8">
               <item>
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Variables</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QTextEdit" name="var"/>
               </item>
              </layout>
             </item>
             <item row="6" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_10">
               <item>
                <widget class="QPushButton" name="add_transition">
                 <property name="text">
                  <string>Add Transition</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="add_state">
                 <property name="text">
                  <string>Add State</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="3" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_7">
               <item>
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>Outputs</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QTextEdit" name="out"/>
               </item>
              </layout>
             </item>
             <item row="2" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Inputs</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QTextEdit" name="in"/>
               </item>
              </layout>
             </item>
             <item row="0" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>Name</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="name">
                 <property name="placeholderText">
                  <string/>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="5" column="0">
              <widget class="QPushButton" name="save_settings">
               <property name="text">
                <string>Save settings</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Comment</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="comment"/>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
           <widget class="QLabel" name="label_10">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>10</y>
              <width>201</width>
              <height>20</height>
             </rect>
            </property>
            <property name="styleSheet">
             <string notr="true">font: bold 16px;</string>
            </property>
            <property name="text">
             <string>FSM Settings</string>
            </property>
           </widget>
          </widget>
          <widget class="QWidget" name="runtime">
           <widget class="QLabel" name="label_9">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>10</y>
              <width>201</width>
              <height>20</height>
             </rect>
            </property>
            <property name="styleSheet">
             <string notr="true">font: bold 16px;</string>
            </property>
            <property name="text">
             <string>Runtime</string>
            </property>
           </widget>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>40</y>
              <width>221</width>
              <height>72</height>
             </rect>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="0">
              <widget class="QComboBox" name="updateInputVariable">
               <property name="toolTip">
                <string>Select input type</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <layout class="QGridLayout" name="gridLayout_2">
               <item row="0" column="0">
                <widget class="QLineEdit" name="updateInputValue">
                 <property name="toolTip">
                  <string>Enter input value</string>
                 </property>
                 <property name="placeholderText">
                  <string>Enter integer value</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QPushButton" name="updateInputButton">
                 <property name="toolTip">
                  <string>Inject the selected input</string>
                 </property>
                 <property name="text">
                  <string>Update</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
           <widget class="QPushButton" name="terminateButton">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>350</y>
              <width>221</width>
              <height>51</height>
             </rect>
            </property>
            <property name="text">
             <string>Terminate</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_11">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>120</y>
              <width>201</width>
              <height>20</height>
             </rect>
            </property>
            <property name="styleSheet">
             <string notr="true">font: bold 14px;</string>
            </property>
            <property name="text">
             <string>Inputs</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_14">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>240</y>
              <width>201</width>
              <height>20</height>
             </rect>
            </property>
            <property name="styleSheet">
             <string notr="true">font: bold 14px;</string>
            </property>
            <property name="text">
             <string>Outputs</string>
            </property>
           </widget>
           <widget class="QTableWidget" name="inputTable">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>140</y>
              <width>221</width>
              <height>91</height>
             </rect>
            </property>
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
            </property>
            <property name="columnCount">
             <number>2</number>
            </property>
            <column/>
            <column/>
           </widget>
           <widget class="QTableWidget" name="outputTable">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>260</y>
              <width>221</width>
              <height>91</height>
             </rect>
            </property>
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
            </property>
            <property name="columnCount">
             <number>2</number>
            </property>
            <column/>
            <column/>
           </widget>
          </widget>
          <widget class="QWidget" name="stateEdit">
           <widget class="QLabel" name="label_7">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>10</y>
              <width>201</width>
              <height>20</height>
             </rect>
            </property>
            <property name="styleSheet">
             <string notr="true">font: bold 16px;</string>
            </property>
            <property name="text">
             <string>State Edit</string>
            </property>
           </widget>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>40</y>
              <width>221</width>
              <height>351</height>
             </rect>
            </property>
            <layout class="QGridLayout" name="gridLayout_5">
             <item row="1" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_9">
               <item>
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Name</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="state_name">
                 <property name="placeholderText">
                  <string/>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="2" column="0">
              <layout class="QGridLayout" name="gridLayout_4">
               <item row="5" column="0">
                <widget class="QPushButton" name="save_state_edit">
                 <property name="text">
                  <string>Save</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>State Action (C++ code)</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QTextEdit" name="state_action"/>
               </item>
               <item row="4" column="0">
                <widget class="QPushButton" name="set_start_state">
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>32</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Set As Start</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="4" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QPushButton" name="back_state_edit">
                 <property name="enabled">
                  <bool>true</bool>
                 </property>
                 <property name="autoFillBackground">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Go Back</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="delete_state">
                 <property name="styleSheet">
                  <string notr="true"/>
                 </property>
                 <property name="text">
                  <string>Delete</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </widget>
          <widget class="QWidget" name="transitionEdit">
           <widget class="QWidget" name="layoutWidget_2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>40</y>
              <width>221</width>
              <height>351</height>
             </rect>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="3" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QPushButton" name="back_transition_edit">
                 <property name="enabled">
                  <bool>true</bool>
                 </property>
                 <property name="autoFillBackground">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Go Back</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="delete_transition">
                 <property name="styleSheet">
                  <string notr="true"/>
                 </property>
                 <property name="text">
                  <string>Delete</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="1" column="0">
              <layout class="QGridLayout" name="gridLayout_7">
               <item row="4" column="0">
                <widget class="QPushButton" name="save_transition_edit">
                 <property name="text">
                  <string>Save</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_12">
                 <property name="text">
                  <string>Condition (bool expresion C++)</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QTextEdit" name="transition_condition"/>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
           <widget class="QLabel" name="label_13">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>10</y>
              <width>201</width>
              <height>20</height>
             </rect>
            </property>
            <property name="styleSheet">
             <string notr="true">font: bold 16px;</string>
            </property>
            <property name="text">
             <string>Transition Edit</string>
            </property>
           </widget>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGraphicsView" name="diagram_view">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>570</width>
            <height>400</height>
           </size>
          </property>
          <property name="sizeIncrement">
           <size>
            <width>1</width>
            <height>1</height>
           </size>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
          </property>
          <property name="interactive">
           <bool>true</bool>
          </property>
          <property name="viewportUpdateMode">
           <enum>QGraphicsView::ViewportUpdateMode::FullViewportUpdate</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextEdit" name="logDisplay"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>Tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="load_act"/>
   <addaction name="save_act"/>
   <addaction name="separator"/>
   <addaction name="start_act"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="load_act">
   <property name="text">
    <string>Load From .fsm</string>
   </property>
   <property name="toolTip">
    <string>Load an FSM file</string>
   </property>
  </action>
  <action name="save_act">
   <property name="text">
    <string>Save To .fsm</string>
   </property>
   <property name="toolTip">
    <string>Save the current FSM</string>
   </property>
  </action>
  <action name="create_act">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Create New</string>
   </property>
   <property name="toolTip">
    <string>Generate C++ code for the FSM</string>
   </property>
  </action>
  <action name="start_act">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Start Automaton</string>
   </property>
   <property name="toolTip">
    <string>Starts the defined fsm</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
